### PR TITLE
Add Filtering Support for Updates

### PR DESCRIPTION
**Pull request by SudharakaP**
Related to https://github.com/opencollective/opencollective/issues/4384

Related Front-end PR: https://github.com/opencollective/opencollective-frontend/pull/6555

### CODE REVIEW

File: server/graphql/v2/interface/Account.js [hunk 1]
```diff
@@ -300,20 +300,56 @@ const accountFieldsDefinition = () => ({
       ...CollectionArgs,
       onlyPublishedUpdates: { type: GraphQLBoolean },
       onlyChangelogUpdates: { type: GraphQLBoolean },
+      sortBy: { type: GraphQLString },
```
Comment (Betree) [inline server/graphql/v2/interface/Account.js:303]: In `ExpensesQuery`, `TransactionsQuery` and `OrdersQuery` it seems that we use `orderBy`, with the `ChronologicalOrderInput` type. Would be great to make it consistent. See https://github.com/opencollective/opencollective-api/blob/b1737a73c4c770a27eefd505749c45cb52d01e2d/server/graphql/v2/query/ExpensesQuery.ts#L105

File: server/graphql/v2/interface/Account.js [hunk 2]
```diff
@@ -300,20 +300,56 @@ const accountFieldsDefinition = () => ({
...
+      searchTerm: { type: GraphQLString },
     },
-    async resolve(collective, { limit, offset, onlyPublishedUpdates, onlyChangelogUpdates }) {
+    async resolve(collective, { limit, offset, onlyPublishedUpdates, onlyChangelogUpdates, sortBy, searchTerm }) {
       let where = {
         CollectiveId: collective.id,
+        [Op.and]: [],
       };
+      const include = [{ association: 'fromCollective', required: true, attributes: [] }];
```
Comment (Betree) [inline server/graphql/v2/interface/Account.js:311]: This `include` does not seem necessary if there's no `searchTerm`

File: server/graphql/v2/interface/Account.js (hunk 1 above)
Comment (SudharakaP) [reply server/graphql/v2/interface/Account.js]: Thanks. Fixed. 👍🏽

File: server/graphql/v2/interface/Account.js (hunk 2 above)
Comment (SudharakaP) [reply server/graphql/v2/interface/Account.js]: Thanks. I've refactored it. 👍🏽

File: server/graphql/v2/interface/Account.js [hunk 3]
```diff
@@ -300,20 +300,52 @@ const accountFieldsDefinition = () => ({
       ...CollectionArgs,
       onlyPublishedUpdates: { type: GraphQLBoolean },
       onlyChangelogUpdates: { type: GraphQLBoolean },
+      orderBy: { type: ChronologicalOrderInput },
```
Comment (Betree) [inline server/graphql/v2/interface/Account.js:303]: I would make it non-nullable with a default value (createdAt DESC). It's good to have a predictive order for things.

Plus, I'm not sure what happens if this is null line 317, probably a crash when we try to do `orderBy.field`?

Comment (SudharakaP) [reply server/graphql/v2/interface/Account.js:303]: Yes, agreed it would crash. Seems I've missed this one. I've just corrected it. 🤔 🐨

**Review by Betree — APPROVED** [review]
Lgtm :+1:

**Pull request merged**